### DISCUSSION TITLE
How can I use morphMap on model_has_roles table ?

### DISCUSSION THREAD

**Discussion by tomohirohiratsuka** [spatie/laravel-permission#1982] [Q&A] [answered]
Hi,

I want to use custom model_type on model_has_roles table with morphMap.
Is there any way to set custom model_type when assign roles to model.

On ServiceProvider
```
Relation::morphMap([
    'user' => 'App\Models\User'
]);
```
On somewhere
```
$user->assignRole('admin'); 
```
I want to store 'user' on model_type field on model_has_roles table at this timing.

**Comment by erikn69** [comment, marked as answer]
Weird thing, i am doing the same and works
![image](https://user-images.githubusercontent.com/4933954/147962498-ac422456-2836-424d-ad3e-7b619546e66e.png)

**Reply by tomohirohiratsuka** [reply]
Sorry I don't get it.
Is the screenshot's model_type filed's value model class name or custom polymorphic name?

I want to set custom polymorphic name which I defined on Relation::morphMap()

**Reply by erikn69** [reply]
Obviously it is the custom polymorphic name, otherwise why would I show it to you? 😄
I'm using manually and https://github.com/spatie/laravel-morph-map-generator on my 2 apps

**Reply by tomohirohiratsuka** [reply]
I figured it out. Thank you.

**Reply by erikn69** [reply]
You probably forgot about `getMorphClass`
```php
class User extends Model
{
    public function getMorphClass(): string {
        return 'user';
    }
}
```
